scripts/provision-kind-cluster.sh

```diff
@@ -45,26 +43,21 @@ if [[ -z "$cluster_name" ]]; then
     exit 1
 fi
 
-shift
-
-# Process our input arguments
-while getopts "b:i:v:k:" opt; do
-  case ${opt} in
-    v ) # K8s version to provision
-        OPTARG="K8_$(echo "${OPTARG}" | sed 's/\./\_/g')"
-        if [ ! -z ${OPTARG+x} ]; then
-            K8_VERSION=${!OPTARG}
-        else
-            echo "K8s version not supported" 1>&2
-            exit 2
-        fi
-      ;;
-    \? )
-        echo "${USAGE}" 1>&2
-        exit
-      ;;
-  esac
-done
+# Process K8_VERSION env var
+
+if [ ! -z ${K8_VERSION} ]; then
+    K8_VER="K8_$(echo "${K8_VERSION}" | sed 's/\./\_/g')"
+    K8_VERSION=${!K8_VER}
+    
+    # Check if version is supported:
+    if [ -z $K8_VERSION ]; then
+        echo "Version set: $K8_VER"
+        echo "K8s version not supported" 1>&2
+        exit 2
+    fi
+else
+    K8_VERSION=${K8_1_16}
+fi
```

Comment:
The above is fine (since it works, of course). However, in a future PR, I'd love to see the above converted into code that is more readable and less Bash-magic.

Perhaps add a function into `scripts/lib/k8s.sh` that takes a version parameter and returns a full SHA1 string from a Bash array lookup (instead of the `sed` and `${!K8_VER}` stuff which I find pretty inscrutable :)

Reply:
totally agree @jaypipes , will work on that at next.